test-data/unit/diff.test

```diff
@@ -948,39 +948,57 @@ __main__.A.g
 [case testFinalFlagsTriggerVar]
 from typing import Final
 
+w: Final = 1
 x: Final = 1
 y: Final[int] = 1
+z: Final[int] = 1
 same: Final = 0
 class C:
+    w: Final = 1
     x: Final = 1
     y: Final[int] = 1
+    z: Final[int] = 1
     same: Final = 0
     def __init__(self) -> None:
-        self.z: Final = 1
-        self.t: Final[int] = 1
+        self.wi: Final = 1
+        self.xi: Final = 1
+        self.yi: Final[int] = 1
+        self.zi: Final[int] = 1
         self.also_same: Final[int] = 0
 
 [file next.py]
 from typing import Final
 
-x = 1
+w = 1
+x: Final[int] = 1
 y: int = 1
+z: Final = 1
 same: Final = 0
 class C:
-    x = 1
+    w = 1
+    x: Final[int]  = 1
     y: int = 1
+    z: Final = 1
     same: Final = 0
     def __init__(self) -> None:
-        self.z = 1
-        self.t: int = 1
-        self.also_same: Final = 0
+        self.wi = 1
+        self.xi: Final[int] = 1
+        self.yi: int = 1
+        self.zi: Final = 1
+        self.also_same: Final[int] = 0
 [out]
-__main__.C.t
+__main__.C.w
+__main__.C.wi
 __main__.C.x
+__main__.C.xi
 __main__.C.y
+__main__.C.yi
 __main__.C.z
+__main__.C.zi
+__main__.w
 __main__.x
```

Comment:
Hm, I am not sure I totally understand why `x: Final = 1` and `x: Final[int] = 1` appear in the diff. Shouldn't we infer exactly the same type `Instance('builtins.int', final_value=Literal[1])` for both of them?

Reply:
No -- we infer `Instance('builtins.int', final_value=None)` for the latter. So, since those two declarations have subtly different behavior, I thought it would be important to test both.

(I also went ahead and expanded this test a little. I forgot to update it to test the new semantics earlier.)

Reply:
Basically, yeah.

In short, I thought it was more important to make sure that statements like `x: Final[float] = 42` and `x: float = 42` behaves in the same way as much as possible. This helps ensure that mypy's behavior stays consistent no matter what, even if we lose track of the original literal value (for example, if the user is messing around with generics).

This does mean that `x: Final[int] = 9` and `x: Final = 9` appear to behave "inconsistently", but I thought that would be less surprising then if `x: Final[int] = 9` and `x: int = 9` were to behave "inconsistently".

Reply:
On second thought though, I wonder if it might be worth making `x: Final[int] = 4` and `x: Final = 4` behave in the same way *only* if the declared and inferred types are equivalent. That way, we don't run into any issues with expressions like `x: Final[float] = 4` since we'd be disabling this "substitution" behavior in that case.

That might be a little too magical though, not sure. LMK if this is an idea you think might be worth exploring.

Reply:
> LMK if this is an idea you think might be worth exploring.

Maybe just open an issue about this. We can decide later when we will have more experience.